File: 程序员面试宝典/1.1.判定字符是否唯一 .py
class Solution:
    def isUnique(self, astr: str) -> bool:
        mask = 0

        for c in astr:
            num = ord(c) - ord('a')
            if mask & (1 << num) != 0:
                return False
            else:
                mask = mask | (1 << num)
        return True


if __name__ == "__main__":
    solu = Solution()
    ans = solu.isUnique('leetcode')
    print(ans)

File: 程序员面试宝典/1.1.判定字符是否唯一 .cpp
#include <string>

using namespace std;

class Solution {
public:
    bool isUnique(string astr)
    {
        int mask = 0;
        for (char c : astr) {
            if (mask & (1 << c - 'a')) {
                return false;
            }
            mask |= (1 << c - 'a');
        }

        return true;
    }
};